src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5028,6 +5028,56 @@ Thread * __stdcall JIT_InitPInvokeFrame(InlinedCallFrame *pFrame, PTR_VOID StubS
 EXTERN_C void JIT_PInvokeBegin(InlinedCallFrame* pFrame);
 EXTERN_C void JIT_PInvokeEnd(InlinedCallFrame* pFrame);
 
+// Forward declaration
+EXTERN_C void STDCALL ReversePInvokeBadTransition();
+
+// This is a slower version of the reverse PInvoke enter function.
+static void JIT_ReversePInvokeEnterRare(FastReversePInvokeFrame* frame)
```

Comment:
We typically mark rare paths like this with `NOINLINE` to make it clear to the C++ compiler that we really want to keep the hot and rare paths separate.